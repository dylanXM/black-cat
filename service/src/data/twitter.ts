import { User } from './user';

const comment = {
  id: 1,
  userId: 1,
  content: 'This is a comment',
  likes: [1, 2, 3, 4, 5],
  time: '9 hours ago',
};

export type Comment = typeof comment;

const twitter = {
  userId: 1,
  channle: 'Channel 1',
  title: 'Activity Title Name Make it Longer May Longer than One Line',
  startTime: '2021-07-01 00:00:00',
  endTime: '2021-07-01 23:59:59',
  content:
    '[No longer than 300 chars] Vivamus sagittis, diam in lobortis, sapien arcu mattis erat, vel aliquet sem urna et risus. Ut feugiat sapien mi potenti[No longer than 300 chars] Vivamus sagittis, diam in lobortis, sapien arcu mattis erat, vel aliquet sem urna et risus. Ut feugiat sapien mi potenti',
  goings: [1, 2, 3, 4, 5],
  likes: [1, 2, 3, 4, 5],
  comments: Array(10)
    .fill(comment)
    .map((c, index) => ({
      ...c,
      id: index,
    })),
  address: 'Address',
  pictures: [
    'https://img.lovepik.com/photo/20211119/medium/lovepik-ten-thousand-mountains-tupian-picture_500348227.jpg',
    'https://img.lovepik.com/photo/20211119/medium/lovepik-ten-thousand-mountains-tupian-picture_500348227.jpg',
    'https://img.lovepik.com/photo/20211119/medium/lovepik-ten-thousand-mountains-tupian-picture_500348227.jpg',
    'https://img.lovepik.com/photo/20211119/medium/lovepik-ten-thousand-mountains-tupian-picture_500348227.jpg',
  ],
};

export type Twitter = typeof twitter & { user?: User };

export const twitterLength = 100;
export const twitterList = Array(twitterLength)
  .fill(twitter)
  .map((t, index) => ({
    ...t,
    id: index,
  }));

export function findTwitterById(id: number) {
  return twitterList.find((item) => item.userId === id);
}

export function findTwitterByChannel(channel: string) {
  return twitterList.find((item) => item.channle === channel);
}

export function findTwitterByUserId(userId: number) {
  return twitterList.filter((item) => item.userId === userId);
}
