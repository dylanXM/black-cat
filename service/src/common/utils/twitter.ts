import { findUserById } from 'src/data/user';
import deepClone from './cloneDeep';

type TypeKey = 'likes' | 'goings';

function patchUsersToTwitter(key: TypeKey, ids: number[]) {
  const users = deepClone(ids.map((id) => findUserById(id)));
  return users;
}

export default patchUsersToTwitter;
