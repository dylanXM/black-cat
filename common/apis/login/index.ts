import { ApiType, request } from '@/common/utils/request';

export interface LoginParams {
  username: string;
  password: string;
}

export async function doLogin(params: LoginParams): Promise<string> {
  const res = await request<ApiType<string>>({
    url: '/auth/login',
    method: 'POST',
    data: params,
  });
  return res.data;
}