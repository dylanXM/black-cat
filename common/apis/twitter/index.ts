import { ApiType, request } from '@/common/utils/request';
import { User } from '../user/user';

export interface GetTwittersParams {
  page: number;
  pageSize: number;
  startDate?: number;
  endDate?: number;
  channel?: string;
}

interface Comment {
  id: number;
  userId: number;
  content: string;
  likes: number[];
  time: string;
}

export interface Twitter {
  id: number;
  userId: number;
  channle: string;
  title: string;
  startTime: string;
  endTime: string;
  content: string;
  goings: number[];
  likes: number[];
  comments: Comment[];
  address: string;
  pictures: string[];
  user?: User;
}

export interface GetTwittersResponse {
  data: Twitter[];
  count: number;
}

export async function getTwitters(params: GetTwittersParams): Promise<GetTwittersResponse> {
  const res = await request<ApiType<GetTwittersResponse>>({
    url: '/twitter/getTwitters',
    method: 'GET',
    params,
  });
  return res.data;
}