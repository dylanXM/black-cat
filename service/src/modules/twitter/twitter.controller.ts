import { Controller, Get, Query, UseGuards } from '@nestjs/common';
import { TwitterService } from './twitter.service';
import { ApiBearerAuth, ApiOperation } from '@nestjs/swagger';
import { JwtAuthGuard } from 'src/common/auth/jwtAuth.guard';
import { GetTwittersDto } from './dto/get-twitter.dto';

@Controller('twitter')
export class TwitterController {
  constructor(private readonly twitterService: TwitterService) {}

  @Get('getTwitters')
  @ApiOperation({ summary: '获取用户个人信息' })
  @UseGuards(JwtAuthGuard)
  @ApiBearerAuth()
  async getTwitters(@Query() params: GetTwittersDto) {
    return await this.twitterService.getTwitters(params);
  }
}
