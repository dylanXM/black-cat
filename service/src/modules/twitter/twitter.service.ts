import { Injectable } from '@nestjs/common';
import { GetTwittersDto } from './dto/get-twitter.dto';
import { twitterList } from 'src/data/twitter';
import { findUserById } from 'src/data/user';

@Injectable()
export class TwitterService {
  async getTwitters(params: GetTwittersDto) {
    console.log('params', params);
    const { page, channel, startDate, endDate, pageSize } = params;

    const startIndex = (page - 1) * pageSize;
    const endIndex = page * pageSize;

    const data = twitterList.filter((item) => {
      if (channel && item.channel !== channel) {
        return false;
      }
      if (startDate && item.time < startDate) {
        return false;
      }
      if (endDate && item.time > endDate) {
        return false;
      }
      return true;
    });

    // 根据userId匹配用户信息
    data.forEach((item) => {
      item.user = findUserById(item.userId);
    });

    const lendth = data.length;

    return {
      data: data.slice(startIndex, endIndex),
      count: lendth,
    };
  }
}
