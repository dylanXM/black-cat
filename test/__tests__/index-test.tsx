import Empty from '@/components/empty';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Comment } from '@/common/apis/twitter';
import ActivityComments from '@/components/activity-comments';



const mockComments: Comment[] = [
  {
    id: 1,
    userId: 1,
    content: 'This is a comment',
    likes: [1, 2, 3, 4, 5],
    time: '9 hours ago',
    user: {
      id: 1,
      username: 'admin',
      password: '',
      email: '123456@123.com',
      avatar:
        'https://yangxm-1317654405.cos.ap-guangzhou.myqcloud.com/ai/tmp_885f4e6f1806cf82aa38d4989f3b0ed5.jpg',
      activityIds: [],
      goingIds: [1, 2, 3],
      likeIds: [7, 8, 9],
    }
  }
];

describe('Index', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Empty text="asdasd" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ActivityComments comments={mockComments} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});