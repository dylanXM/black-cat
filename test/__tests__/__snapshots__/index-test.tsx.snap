// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={60}
    bbWidth={60}
    data-name="Layer 1"
    fill="#D3C1E5"
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "height": 60,
          "width": 60,
        },
        {
          "flex": 0,
          "height": 60,
          "width": 60,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4292067813,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M16 32A16 16 0 0 1 2.17 8 16 16 0 0 1 3.5 6 16 16 0 1 1 16 32m14-16a14 14 0 0 0-.16-1.85A13.5 13.5 0 0 0 28 14a13.86 13.86 0 0 0-6.23 1.52 24 24 0 0 1 4.64 9.75A13.9 13.9 0 0 0 30 16m-5.3 10.91a22.3 22.3 0 0 0-4.64-10.41A14 14 0 0 0 14 28a14 14 0 0 0 .15 1.84A14 14 0 0 0 16 30a13.87 13.87 0 0 0 8.7-3.09M12.11 29.4c0-.46-.09-.93-.09-1.4a16 16 0 0 1 6.72-13A23 23 0 0 0 17 13.33 16 16 0 0 1 4 20c-.47 0-.92 0-1.38-.09a14 14 0 0 0 9.49 9.49M4.47 8.09A13.9 13.9 0 0 0 2 16a14 14 0 0 0 .19 1.86A13.5 13.5 0 0 0 4 18a14 14 0 0 0 11.39-5.91 22.3 22.3 0 0 0-10.92-4m13.4-5.91A14 14 0 0 0 16 2 13.94 13.94 0 0 0 6 6.28a24.4 24.4 0 0 1 10.45 4.09A13.85 13.85 0 0 0 18 4a14 14 0 0 0-.12-1.81Zm2.06.44c.07.46.07.91.07 1.38a15.9 15.9 0 0 1-2 7.68 25 25 0 0 1 2.36 2.24A15.9 15.9 0 0 1 28 12c.47 0 .94 0 1.4.09a14 14 0 0 0-9.47-9.47"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        fillRule={0}
        matrix={
          [
            1,
            0,
            0,
            1,
            -0.02,
            0,
          ]
        }
        propList={
          [
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "#BABABA",
        "marginTop": 11,
      }
    }
  >
    asdasd
  </Text>
</View>
`;

exports[`Index renders correctly 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={60}
    bbWidth={60}
    data-name="Layer 1"
    fill="#D3C1E5"
    focusable={false}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "height": 60,
          "width": 60,
        },
        {
          "flex": 0,
          "height": 60,
          "width": 60,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4292067813,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M16 32A16 16 0 0 1 2.17 8 16 16 0 0 1 3.5 6 16 16 0 1 1 16 32m14-16a14 14 0 0 0-.16-1.85A13.5 13.5 0 0 0 28 14a13.86 13.86 0 0 0-6.23 1.52 24 24 0 0 1 4.64 9.75A13.9 13.9 0 0 0 30 16m-5.3 10.91a22.3 22.3 0 0 0-4.64-10.41A14 14 0 0 0 14 28a14 14 0 0 0 .15 1.84A14 14 0 0 0 16 30a13.87 13.87 0 0 0 8.7-3.09M12.11 29.4c0-.46-.09-.93-.09-1.4a16 16 0 0 1 6.72-13A23 23 0 0 0 17 13.33 16 16 0 0 1 4 20c-.47 0-.92 0-1.38-.09a14 14 0 0 0 9.49 9.49M4.47 8.09A13.9 13.9 0 0 0 2 16a14 14 0 0 0 .19 1.86A13.5 13.5 0 0 0 4 18a14 14 0 0 0 11.39-5.91 22.3 22.3 0 0 0-10.92-4m13.4-5.91A14 14 0 0 0 16 2 13.94 13.94 0 0 0 6 6.28a24.4 24.4 0 0 1 10.45 4.09A13.85 13.85 0 0 0 18 4a14 14 0 0 0-.12-1.81Zm2.06.44c.07.46.07.91.07 1.38a15.9 15.9 0 0 1-2 7.68 25 25 0 0 1 2.36 2.24A15.9 15.9 0 0 1 28 12c.47 0 .94 0 1.4.09a14 14 0 0 0-9.47-9.47"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        fillRule={0}
        matrix={
          [
            1,
            0,
            0,
            1,
            -0.02,
            0,
          ]
        }
        propList={
          [
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "#BABABA",
        "marginTop": 11,
      }
    }
  >
    asdasd
  </Text>
</View>
`;
